x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -140,10 +185,16 @@ static Number negate(Number n) {
             }
             return Float.valueOf(-n.floatValue());
         }
+        if (n instanceof BigInteger) {
+            if (((BigInteger) n).signum() != 0) {
```

Comment:
To this point a BigInteger can only be `>= 0`. So the check is there for the `> 0` part, since negating a non-negative unsigned long should trigger an error response. I.e. `!= 0`.